compiler/hash-ast/src/ast.rs

```diff
@@ -683,14 +677,24 @@ pub struct IgnorePat;
 /// A pattern. e.g. `Ok(Dog {props = (1, x)})`.
 #[derive(Debug, PartialEq, Clone)]
 pub enum Pat {
+    /// An access pattern is one that denotes the access of a property from
+    /// another pattern. This is used to denote namespace accesses like
+    /// `a::b::c`
+    Access(AccessPat),
+    /// A simple binding pattern, assign some value to the name of the pattern
+    Binding(BindingPat),
+    /// A representation of a constructor in the pattern space. Constructors in
+    /// patterns can either be enum or struct values. The subject of the
+    /// constructor can be either an [Pat::Access] or a [Pat::Binding].
     Constructor(ConstructorPat),
+    /// Namespace pattern is used to destructure entries from an import.
     Namespace(NamespacePat),
```

Comment:
I think we should rename this to ModulePat

Reply:
I had the same thought 😁